nuxeo-drive-client/nxdrive/engine/watcher/remote_watcher.py

```diff
@@ -164,24 +164,33 @@ def _scan_pair(self, remote_path):
 
     @staticmethod
     def _check_modified(child_pair, child_info):
-        if (child_pair.remote_can_delete != child_info.can_delete
-                or child_pair.remote_can_rename != child_info.can_rename
-                or child_pair.remote_can_update != child_info.can_update
-                or child_pair.remote_can_create_child != child_info.can_create_child
-                or child_pair.remote_digest != child_info.digest):
-            return True
-        return False
+        return any([
+            child_pair.remote_can_delete != child_info.can_delete,
+            child_pair.remote_can_rename != child_info.can_rename,
+            child_pair.remote_can_update != child_info.can_update,
+            child_pair.remote_can_create_child != child_info.can_create_child,
+            child_pair.remote_digest != child_info.digest,
+        ])
+
+    def _do_scan_remote(self, doc_pair, remote_info, force_recursion=True, moved=False):
+        # NXDRIVE-881: check the folder was not remotely renamed.
+        # Note: local_name = '' for the root folder, so skip it.
+        local_name = os.path.basename(doc_pair.local_path)
+        if local_name and local_name != remote_info.name:
+            self._dao.force_remote(doc_pair)
+            self._dao.add_path_to_scan(
+                self._init_scan_remote(doc_pair, remote_info))
```

Comment:
Why not just `self._dao.add_path_to_scan(remote_info.path)`? I think that `_init_scan_remote` will then always be called in either `_scan_remote_scroll` or `_scan_remote_recursive`, though it might be interesting to check at this point, not sure...